test/cli/other_test.py

```diff
@@ -2276,3 +2277,49 @@ def test_dumpfile_platform(tmpdir):
                 break
     assert ' wchar_t_bit="' in platform
     assert ' size_t_bit="' in platform
+
+
+def test_builddir_hash_check_level(tmp_path):  # #13376
+    test_file = tmp_path / 'test.c'
```

Comment:
I have the feeling that a test in TestCppcheck could be easier.

It's important that we have some pytest that test the whole chain and checks that the file is rechecked if the hash is changed but I assume we already have that.

Reply:
> I have the feeling that a test in TestCppcheck could be easier.

I do not think this can be tested in a unit tests as it requires output files to be generated. Also the logic for the hash generation lives within `CppCheck` and I do not think we should expose that logic via interface.

> It's important that we have some pytest that test the whole chain and checks that the file is rechecked if the hash is changed but I assume we already have that.

It can probably be simplified or merge with some other tests but that probably holds true for several of those tests. Will take a look when I add more builddir-related tests in the future.